### PR TITLE
Move SEV ioctls over to iocuddle

### LINKED ISSUE #112: Update SEV ioctls to be iocuddle-based types
- [x] PlatformReset
- [x] PlatformStatus
- [x] PekGenerate
- [x] PekCertificateSigningRequest
- [x] PdhGenerate
- [x] PdhCertificateExpor
- [x] PekCertificateImport
- [x] GetIdentifier

### PR DESCRIPTION
**Pull request by connorkuehl**
Closes #112 

My redemption arc

### CODE REVIEW

File: sev/src/firmware/linux/ioctl.rs [hunk 1]
```diff
@@ -80,14 +80,28 @@ pub struct Command<'a, T: Code> {
...
+    /// platform/kernel from writing to the caller's address space if it wants to. This is primarily
+    /// a semantic tool for programming against the SEV ioctl API.
+    pub fn readonly(subcmd: &'a T) -> Self {
+        Command {
+            code: T::CODE,
+            data: subcmd as *const T as u64,
+            error: 0,
+            _phantom: PhantomData,
+        }
+    }
```
Comment (npmccallum) [inline sev/src/firmware/linux/ioctl.rs:106]: I think the idiomatic names we want here are `from()` and `from_mut()`.

File: sev/src/firmware/types.rs [hunk 2]
```diff
@@ -92,3 +92,29 @@ impl<'a> PekCsr<'a> {
...
+///
+/// (Chapter 5.11)
+#[repr(C, packed)]
+pub struct PdhCertExport<'a, 'b> {
+    pdh_addr: u64,
+    pdh_len: u32,
+    certs_addr: u64,
+    certs_len: u32,
+    _phantom0: PhantomData<&'a ()>,
+    _phantom1: PhantomData<&'b ()>,
```
[the remark this answers is not in the document]
Comment (connorkuehl) [reply sev/src/firmware/types.rs]: You are correct, I've removed the unnecessary lifetime.

File: sev/src/firmware/types.rs [hunk 3]
```diff
@@ -88,6 +88,32 @@ impl<'a> PekCsr<'a> {
...
+///
+/// (Chapter 5.9; Table 29)
+#[repr(C, packed)]
+pub struct PekCertImport<'a, 'b> {
+    pek_addr: u64,
+    pek_len: u32,
+    oca_addr: u64,
+    oca_len: u32,
+    _phantom0: PhantomData<&'a ()>,
+    _phantom1: PhantomData<&'b ()>,
```
[the remark this answers is not in the document]
Comment (connorkuehl) [reply sev/src/firmware/types.rs]: No. Removed redundant lifetime.

File: sev/src/firmware/types.rs [hunk 4]
```diff
@@ -144,3 +144,30 @@ impl<'a, 'b> PdhCertExport<'a, 'b> {
...
+    id_addr: u64,
+    id_len: u32,
+    _phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> GetId<'a> {
+    pub fn new(id: &'a mut [u8; 64]) -> Self {
+        Self {
+            id_addr: id as *mut _ as _,
+            id_len: std::mem::size_of_val(id) as _,
```
[the remark this answers is not in the document]
Comment (connorkuehl) [reply sev/src/firmware/types.rs]: Yikes! Good catch. Updated.

File: sev/src/firmware/types.rs [hunk 5]
```diff
@@ -144,3 +144,30 @@ impl<'a, 'b> PdhCertExport<'a, 'b> {
...
+        }
+    }
+
+    /// Create a Vec of bytes from the GetId data buffer. This method is only
+    /// meaningful if the data buffer has been populated via the GET_ID2 ioctl
+    /// call because the kernel will modify the `GetId.id_len` field with the
+    /// actual length of the CPU's ID.
+    pub unsafe fn to_vec(&self) -> Vec<u8> {
+        std::slice::from_raw_parts(self.id_addr as *const u8, self.id_len as _).to_vec()
+    }
```
[the remark this answers is not in the document]
Comment (connorkuehl) [reply sev/src/firmware/types.rs]: I agree. Updated with the latter suggestion.

**Pull request merged**